src/transformers/generation_tf_utils.py

```diff
@@ -911,7 +916,7 @@ def _generate_no_beam_search(
             if eos_token_id is not None and cur_len < min_length:
                 # create eos_token_id boolean mask
                 is_token_logit_eos_token = tf.convert_to_tensor(
-                    [True if token is eos_token_id else False for token in range(vocab_size)], dtype=tf.bool
+                    [True if token == eos_token_id else False for token in range(vocab_size)], dtype=tf.bool
```

Comment:
Using `token is eos_token_id` inside the list comprehension is not working well, as demonstrated in the example below. Using `token == eos_token_id` works fine. 

```
eos_token_id = 257  # If changed to <= 256, there is no problem
vocab_size = 257 + 1

# This gives `[False, False, False]`
print([True if token is eos_token_id else False for token in range(vocab_size)][-3:])

token = 257
# This gives `True`
print(token is eos_token_id)
```

Reply:
Agree that `is` is totally wrong for equality testing here. This was weird enough that I checked the codebase to make sure a similar pattern wasn't showing up anywhere else, and `generation_tf_utils` has the only occurrences.